### ISSUE TITLE
Stronger warning on test -z

### ISSUE THREAD

**Issue by rowlap** [koalaman/shellcheck#1353]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash 
somestring=$(ls /data)

if [ -z $somestring ]; then
  echo somestring is empty
else
  rm -rf /data/$somestring
fi
```

#### Here's what shellcheck currently says:
```
Line 4	SC2086: Double quote to prevent globbing and word splitting.
Line 7	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

Because the shell will elide `$somestring` if undefined or empty, the `if` command is rendered
`if '[' '-z' ']'; then`
creating the 1-arg version of test, with `'-z'` being the string under test, which then returns success (0).

It *appears* superficially that the code is working, because empty/missing strings pass the test, but not in the intended sense of test evaluating the length of the second arg.

Additionally, if `$somestring` contains whitespace, test will fail with syntax error ("binary operator expected" or "extra argument" from GNU coreutils test, "too many arguments" from bash).

#### Proposal:

shellcheck already identifies SC2086 and possibly SC2154 for the -z line. Please add a new warning or error specifically to handle test -z with unquoted variable expansion.

**Comment by rowlap** [comment]
This might be a duplicate of issue #750 - certainly if a fully general solution to quoting test arguments was implemented, this `-z` handling issue would be covered.

**Comment by koalaman** [comment]
I wrote a whole blog post about this at one point: https://www.vidarholen.net/contents/blog/?p=576

I don't actually agree that `-z` specifically should be stronger -- if anything it's less likely to matter -- but there is some argument to be made that variables in `[ .. ]` should have stronger warnings in general because you're less likely to want word splitting.
